app/models/multi_record_csv.rb

```diff
@@ -1,64 +1,72 @@
-class MultiRecordCsv
+# frozen_string_literal: true
 
+class MultiRecordCsv
   attr_reader :collections
-  attr_reader :granules
-
-  IDENTIFIER_COLUMN = "DataSetId (ShortName) - umm-json link"
-  GRANULE_IDENTIFIER_COLUMN = "Collection Name (Concept Id) - Granule UR"
-
-  METRIC_FIELDS     = [
-    "# Red fields (absolute errors):",
-    "# Yellow fields (recommended fixes)",
-    "# Blue fields (observations/ may or may not need to be fixed)",
-    "# np fields (not in the metadata, and not marked by any color)",
-    "# fields checked",
-    "% red fields",
-    "% yellow fields",
-    "% blue fields"
-  ]
+
+  METRIC_FIELDS = [
+    '# Red fields (absolute errors):',
+    '# Yellow fields (recommended fixes)',
+    '# Blue fields (observations/ may or may not need to be fixed)',
+    '# np fields (not in the metadata, and not marked by any color)',
+    '# fields checked',
+    '% red fields',
+    '% yellow fields',
+    '% blue fields'
+  ].freeze
   FORMATS = Collection::SUPPORTED_FORMATS
 
   def initialize(records)
-    @collections = records.where(recordable_type: "Collection")
-    @granules    = records.where(recordable_type: "Granule")
+    @collections = records.where(recordable_type: 'Collection')
   end
 
   def to_csv
     CSV.generate do |csv|
-      csv << ["CMR Multiple Record Report"]
+      csv << ['CMR Multiple Record Report']
 
       if collections.any?
-        csv << ["Collections"]
-
         # Collections
         FORMATS.each do |metadata_format|
           records_for_format = collections.metadata_format(metadata_format)
           next if records_for_format.empty?
 
-          fields = determine_fields(records_for_format)
+          granules = []
+          # Get a list of all granules
+          records_for_format.each do |record|
+            associated_granule_value = record.associated_granule_value
+            next unless !associated_granule_value.nil? &&
```

Comment:
This is kind of confusing because you are using `unless` and a lot of `!`s. Is it possible to use `if` and make it a bit more straightforward?

Reply:
I originally had if with if but ruby cop changed it :)

Reply:
maybe because next in assumed to be the natural state and rubocop wants to focus on the exception? I vote for rebelling against our robot overlords and change it to if

Reply:
alright I should probably also change it from nil? to empty? anyhow